rust/nasl-builtin-network/src/socket.rs

```diff
@@ -133,9 +147,17 @@ pub struct NaslSockets {
 }
 
 impl NaslSockets {
-    fn open_udp(addr: &str, port: u16) -> Result<NaslSocket, FunctionErrorKind> {
-        let socket = UdpSocket::bind("0.0.0.0:0")?;
-        socket.connect(format!("{addr}:{port}"))?;
+    fn open_udp(addr: IpAddr, port: u16) -> Result<NaslSocket, FunctionErrorKind> {
+        let sock_addr = (addr, port).to_socket_addrs()?.next().ok_or_else(|| {
+            FunctionErrorKind::Diagnostic(
+                format!(
+                "the given address and port do not correspond to a valid address: {addr}:{port}",
+            ),
+                None,
+            )
+        })?;
```

Comment:
Since this is used multiple times: Extract into `resolve_sock_addr(addr, port)`?